Rename a few RPCs to improve clarity

tiyash-basu-frequenz opened the pull request:
The RPC being renamed here are the ones that fetch microgrid and component metadata, and live component data.

The following RPC has been renamed to make their objectives clearer:
  * `GetMetadata` -> `GetMicrogridMetadata`
  * `ListComponents` -> `ListComponentMetadata`
  * `GetComponentData` -> `StreamComponentData`

tiyash-basu-frequenz committed:
Rename `microgrid.Metadata` to `microgrid.MicrogridMetadata`

This improved the name of the message, making it clearer that it represents
the metadata of the microgrid.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

leandro-lucarella-frequenz commented on RELEASE_NOTES.md:
```diff
@@ -17,6 +17,8 @@
   microgrid metadata. The returned value is an instance of the message
   `Metadata`.
 
+  Note that this RPC has been renamed to `GetMicrogridMetadata` in [this PR](https://github.com/frequenz-floss/frequenz-api-microgrid/pull/43).
```
If this is a new feature, I would just update the previous release notes item to use the final name.

tiyash-basu-frequenz replied:
Done 🚀

leandro-lucarella-frequenz reviewed (dismissed):
Besides the very small improvement suggestion for the release notes, LGTM!

tiyash-basu-frequenz committed:
Rename a few RPCs to improve clarity

The RPC being renamed here are the ones that fetch microgrid and component
metadata, and live component data.

The following RPC has been renamed to make their objectives clearer:
  * `GetMetadata` -> `GetMicrogridMetadata`
  * `GetComponentData` -> `StreamComponentData`

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

# Conflicts:
#	RELEASE_NOTES.md
